UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs

```diff
@@ -9,56 +9,51 @@ namespace Testing.Common.ActiveDirectory
+        public static async Task RemoveTheUserFromTheGroup(string user, string groupId, string token)
         {
-            using (var client = new HttpClient())
-            {
-                client.DefaultRequestHeaders.Authorization =
-                    new AuthenticationHeaderValue("Bearer", token);
-                var httpRequestMessage = new HttpRequestMessage(HttpMethod.Delete,
-                    $@"https://graph.microsoft.com/v1.0/groups/{groupId}/members/{user}/$ref");
-                var result = client.SendAsync(httpRequestMessage).Result;
-                return result.IsSuccessStatusCode;
-            }
+            var url = $@"{ApiBaseUrl}/groups/{groupId}/members/{user}/$ref";
+            await SendGraphApiRequest(url, HttpMethod.Delete, token);
+            Console.WriteLine($"Deleted group '{groupId}' from user: {user}");
         }
 
-        public static async Task<bool> DeleteTheUserFromAd(string user, string token)
+        public static async Task DeleteTheUserFromAd(string user, string token)
+        {
+            var url = $@"{ApiBaseUrl}/users/{user}";
+            await SendGraphApiRequest(url, HttpMethod.Delete, token);
+            Console.WriteLine($"Deleted user: {user}");
+        }
+
+        private static async Task<string> SendGraphApiRequest(string url, HttpMethod method, string token)
         {
-            
-            var tenantId = TestConfig.Instance.AzureAd.TenantId;
             var policy = Policy.HandleResult<HttpResponseMessage>(r => r.StatusCode == HttpStatusCode.NotFound)
-                .WaitAndRetryAsync(3, retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)));
+                .WaitAndRetryAsync(3, retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)),
+                    (msg, time) => { Console.WriteLine($"Received {msg.Result.StatusCode} for {method} {url}"); });
            
-            
             // sometimes the api can be slow to actually allow us to access the created instance, so retry if it fails the first time
             var result = await policy.ExecuteAsync(async () =>
             {
                 using (var client = new HttpClient())
                 {
                     client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
-                    var httpRequestMessage = new HttpRequestMessage(HttpMethod.Delete,
-                    $@"https://graph.microsoft.com/v1.0/{tenantId}/users/{user}");
+                    var httpRequestMessage = new HttpRequestMessage(method, url);
                     return await client.SendAsync(httpRequestMessage);
                 }
             });
-            
-            return result.IsSuccessStatusCode;
-            
+
+            var response = await result.Content.ReadAsStringAsync();
+            if (!result.IsSuccessStatusCode)
+                throw new Exception($"Failed to execute {method} on {url}, got response {result.StatusCode}: {response}");
```

Comment:
You are throwing exception if not successful status code i.e. not in this range (200-299), which means your retry policy will NOT retry as you have said HandleResult, this will be exception, no a result

Reply:
Yep. Bubble any errors so we can fix them.